### PR TITLE
feat: add a bunch of organization-wide labels

### PR DESCRIPTION
**Pull request by kdmccormick**
## Description

This populates labels.yml with a list of labels that will be synced across the openedx GitHub organization. The labels will be synced by Axim using repo_checks.py.

The labels' spelling, color, and description will be made consistent. If a repo has an existing label with similar spelling (that is: ignoring capitalization, emoji, spaces, and special characters) then the label will be edited in-place instead of created anew. So, we shouldn't have a bunch of old-spelling labels lying around when this is done.

The proposed labels.yml is based on the [comments on this wiki page](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3695214629/openedx+org-wide+labels#2.-Proposal%3A-List-of-starter-labels).

### Blocking work

This PR is based on the work from https://github.com/openedx/terraform-github/pull/60, which actually implements the labels.yml syncing. If you want to review the _implementation_ rather than the _labels_, refer to that PR.

Also, before applying these labels to the org, we need to ensure that:
* the OSPR bot is no longer managing these labels in openedx repos.
* the DEPR workflow is no longer trying to add the `DEPR` label to repos.

### Followup work

Once `core contributor` is added to every repo, we can wrap up: https://github.com/openedx/openedx-webhooks/issues/227

### No emoji

I decided not to include emoji in the labels. My driving motivation is that searches/filters become harder to type and harder to read when emoji are involved. Consider:

```
org:openedx is:open label:epic,initiative,discovery 
```

versus

```
org:openedx is:open label:":dizzy: epic",":100: initiative",":mag_right: discovery"
```

I could be persuaded otherwise; what's most important to me is that we're consistent :)

### Things to keep in mind...

* This is a first pass. We can edit these new labels later. We don't have the machinery to _delete_ labels yet, but we could certainly add that later too if necessary. _Anything is better than what we have now._
* We don't know exactly how these labels should be used, and even if we did, we don't have time to police the community to make sure they're used "correctly". So, rather than trying to design a single perfect workflow, I've tried to provide a set of generally useful set of labels that we can experiment with.
* Any proposed changes to the CONTRIBUTION WORKFLOW labels should involve @mphilbrick211 and @itsjeyd , as they are the ones that run that process.

### CODE REVIEW

**Comment by kdmccormick** [comment]
FYI @e0d

File: migrate/labels.yml
Comment (kdmccormick) [inline migrate/labels.yml:122]: @nedbat Would you rather have `blended` be added here, or would you rather continue letting the OSPR bot do that?

File: migrate/labels.yml
Comment (brian-smith-tcril) [inline migrate/labels.yml:22]: :heart: the friendly color names in here! Amazing documentation of the label use via descriptions, and also the comments for categories! This is just wonderful!

File: migrate/labels.yml
Comment (brian-smith-tcril) [inline migrate/labels.yml:35]: Would code cleanup/refactoring fit under this? I feel like codebase health fits under platform health, but I'm curious as to how others interpret this label.

File: migrate/labels.yml
Comment (brian-smith-tcril) [inline migrate/labels.yml:92]: Could we use "author inactive" or "author unresponsive" here? I feel like "inactive" is a little ambiguous as to what/who is inactive.

**Review by brian-smith-tcril — APPROVED** [review]
These are wonderful, and the `labels.yml` file is extremely easy to read. This makes me super happy!

I left a few comments in there, mostly hoping to start a few conversations. Even if none of the comments I made are addressed I'm happy to say :shipit:!

File: migrate/labels.yml
Comment (nedbat) [inline migrate/labels.yml:27]: Who will be adding this label to issues?  Sort of by default, all issues are "help wanted".  Is there some specific kind of ready this is meant to indicate other than it's an open issue?

File: migrate/labels.yml
Comment (nedbat) [reply migrate/labels.yml:92]: As with all labels, I'm wondering who is adding this label, and who is looking for it?  Whoever is marking a PR as "inactive", why wouldn't they close the PR?  Is the label meant to be bookkeeping on closed PRs?  Would we ever have an open PR with the "inactive" label?

Comment (nedbat) [reply migrate/labels.yml:92]: ... and I just saw "closed inactivity" below.

Comment (kdmccormick) [reply migrate/labels.yml:92]: Each label in the `LABELS USED IN THE OPEN EDX CONTRIBUTION WORKFLOW` section already exists in any repo that the OSPR bot has interacted with (which is most repos). So, these labels are not adding anything new, they are just pulling what the OSPR bot has already been doing in labels.yml.

> Whoever is marking a PR as "inactive", why wouldn't they close the PR? Is the label meant to be bookkeeping on closed PRs? Would we ever have an open PR with the "inactive" label?

@mphilbrick211 and @itsjeyd, who use all the labels in this section, explained how the labels are used [on the italicized notes on this table](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3695214629/openedx+org-wide+labels#2.-Proposal%3A-List-of-starter-labels).  I think that I will copy those notes into the YAML file as comments.

[1] "most" being any repo that the OSPR bot has interacted with, so any repo with non-2U/Axim contributions.

File: migrate/labels.yml
Comment (kdmccormick) [reply migrate/labels.yml:27]: @nedbat To me, the `help wanted` label is meant to answer the question "I'd like to help with this group/project, what should I work on next?" Consider it a superset of `good first issue`.

Open issues that _wouldn't_ warrant a `help wanted` labels are:
* a "draft" issue; one that hasn't been refined well, maybe something you wrote up quickly with the intent of fleshing out later, like [this one](https://github.com/openedx/edx-platform/issues/31768).
* issues that require special permissions, like Axim requests, or certain WG tickets [like this one](https://github.com/openedx/wg-security/issues/29).
* ones that require special context [like this one](https://github.com/openedx/openedx-k8s-harmony/issues/28).
* issues with a high technical barrier to entry. 
* issues that are part of an actively  funded project (ie, we are already paying someone to do it).
* issues that already have intended assignees, like [this one](https://github.com/openedx/wg-build-test-release/issues/279).

> Who will be adding this label to issues? 

Anyone! I'll be using it for my working group, and I'll encourage other WG leads to as well.

File: migrate/labels.yml
Comment (kdmccormick) [reply migrate/labels.yml:35]: Good question, I'm torn it this. When I think of `maintenance` I think moreso of upgrades or other keeping-the-lights-on work, rather than proactive improvements. Do we want a separate label like `refactoring`, `platform-health`, or `technical-investment` ? (or `tech-debt`, but I've come to feel that the term has developed an unfortunate connotation of "let's deprioritize this").

File: migrate/labels.yml
Comment (nedbat) [reply migrate/labels.yml:27]: This sort of guidance will be good to have somewhere.  This file is not discoverable, but I'm not sure where else to put it.

Comment (kdmccormick) [reply migrate/labels.yml:27]: OK, maybe this is something we could add to maintainer playbooks or contribution guides on docs.openedx.org.

That said, I am hoping that the labels can be as self-explanatory as possible. We can't assume potential new contributors will be reading docs.openedx.org, so the more obvious we can make ourselves to the outside world with the tools GitHub gives us, the better.

Is it not reasonably clear what `help wanted` means?

Comment (nedbat) [reply migrate/labels.yml:27]: My only concern about "help wanted" is the interpretation that if the label isn't on a PR, then we don't want or need help with it.

My suggestion about putting your guidelines somewhere was not so that new contributors would find them, but so that we who are labelling issues would have a common understanding of the labels.

**Comment by wittjeff** [comment]
Labels that I have used across projects at edX: 
- a11y (accessibility-related)
- WCAG (is a WCAG conformance issue; not all a11y issues are WCAG issues)
- WCAG 2.0, WCAG 2.1, WCAG 2.2 (specific WCAG standard levels)
I also tag some things 'WCAG AAA' (refers to specific WCAG standards that are above widely accepted legal minimum) but I don't use that very often.

File: migrate/labels.yml
Comment (brian-smith-tcril) [reply migrate/labels.yml:35]: Of those suggestions I like `refactoring` and `technical-investment` best. I'm not sure about `platform-health`, I'm not sure what would make me like it more, maybe `codebase-health`?

Overall I think I'm leaning towards `refactoring`.

Comment (kdmccormick) [reply migrate/labels.yml:35]: 👍🏻

File: migrate/labels.yml
Comment (kdmccormick) [reply migrate/labels.yml:27]: > My suggestion about putting your guidelines somewhere was not so that new contributors would find them, but so that we who are labelling issues would have a common understanding of the labels.

That makes sense. I think the maintainership playbooks would be a good home for these docs, then.

File: migrate/labels.yml
Comment (nedbat) [reply migrate/labels.yml:35]: I keep coming back to "who will apply the label" and "who will read the label".  I don't understand what the separate labels like "refactoring" will do? How will they help people find issues/PRs?

File: migrate/labels.yml
Comment (itsjeyd) [inline migrate/labels.yml:15]: @kdmccormick It might be a temporary issue but this site was unreachable for me when I tried to access it today.

If necessary, https://www.color-hex.com/ might be a good alternative to mention here.

File: migrate/labels.yml
Comment (itsjeyd) [inline migrate/labels.yml:86]: @kdmccormick I added a [section](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3548807177/Community+Contributions+Project+Manager#Current-OSPR-Project-Managers) listing current holders of the OSPR Project Manager role to the wiki page on OSPR management.

Perhaps it would be helpful to link to that here, so whoever reads this can easily look up who to ping?

CC @mphilbrick211

**Review by itsjeyd — COMMENTED** [review]
This looks great @kdmccormick, it'll be great to have a standard set of org-wide labels.

File: migrate/labels.yml
Comment (brian-smith-tcril) [reply migrate/labels.yml:118]: another wording option: "PR **author** is a CC..."

File: migrate/labels.yml
Comment (brian-smith-tcril) [reply migrate/labels.yml:35]: > I keep coming back to "who will apply the label" and "who will read the label". I don't understand what the separate labels like "refactoring" will do? How will they help people find issues/PRs?

I guess my thought on the split between `maintenance` and `refactoring` is `maintenance` implies "if we don't do this things will break," where as `refactoring` implies "this will be nice for making the codebase easier to work in."

With that in mind, I would want to make sure that any `maintenance` PRs were reviewed/merged quite quickly, where as `refactoring` PRs could stay in the review state and gather feedback for a bit longer.

File: migrate/labels.yml
Comment (kdmccormick) [reply migrate/labels.yml:118]: @itsjeyd I haven't used the label at all myself. What I have listed in the description matches the intended behavior of the automatically-added `core committer` label, although that stopped working reliably a year or two ago.

As far as I can tell, the label exists to smooth the contribution workflow for CCs. What do you and @mphilbrick211 think is the best behavior? Whatever it is, I'm happy to describe that here. If the label seems unnecessary, that is OK too; we can remove it.

File: migrate/labels.yml
Comment (nedbat) [reply migrate/labels.yml:35]: Wouldn't the `refactoring` label be doing the same job as the `refactor:` conventional commit category?

Comment (kdmccormick) [reply migrate/labels.yml:35]: Sure, much like `bug` is doing the same job as `fix:`, and `enhancement` is doing the same job as `feature:`, and `documentation` is doing the same job as `docs:`.

Unresolved issues don't have commits, though, so we can't use conventional commit types to label them.

Perhaps each conventional commit type should map to a label? @nedbat , what do you think?

Comment (brian-smith-tcril) [reply migrate/labels.yml:35]: I'd expect a lot of `maintenance` PRs to have commits that go under `chore:`. I think my main point is I'd like to be able to "filter by label" and see how many PRs need somewhat immediate attention because they have the `maintenance` label without needing to look at additional information.

I think in general I'm less hoping to have a `refactor` label than hoping to not have refactoring PRs clutter up the `maintenance` label.

File: migrate/labels.yml
Comment (mphilbrick211) [reply migrate/labels.yml:118]: @itsjeyd @kdmccormick for the purposes of the OSPRs on the Contributions board, I have been using it if the author of the PR is a CC (and I've seen CCs who add that label for that reason as well). Typically if I see that a CC is the author of a PR, it lets me know that that PR will generally be handled in a timely manner, and I may not check in on those as frequently as others. This is because other CCs tend to jump in and help provide feedback and review on those, and the process tends to move fairly quickly.

I agree that it would be good to solidify what that label is used for so it's not confusing. 

On a related note, is there a way to tell which CCs have rights in each repo?

Comment (itsjeyd) [reply migrate/labels.yml:118]: @mphilbrick211 

> On a related note, is there a way to tell which CCs have rights in each repo?

I'm not sure if we have a single source of truth that's always up-to-date for each CC. (I've mainly been referencing the "Project(s) Working On" column from [this wiki page](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3156344833/Core+Contributors+to+the+Open+edX+Project) when trying to check but the info listed there isn't exhaustive for each CC and doesn't follow a consistent format.)

If I understand correctly, that was one of the reasons for retiring the workflow that would add the old `core committer` label to PRs automatically.

CC @kdmccormick

Comment (kdmccormick) [reply migrate/labels.yml:118]: @itsjeyd @mphilbrick211 Sounds good! I'll update the description to indicate that `core contributor` means that the author is generally a CC, even if they don't have rights to the specific repo against which the PR is open.

> I'm not sure if we have a single source of truth that's always up-to-date for each CC. (I've mainly been referencing the "Project(s) Working On" column from [this wiki page](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3156344833/Core+Contributors+to+the+Open+edX+Project) when trying to check but the info listed there isn't exhaustive for each CC and doesn't follow a consistent format.)

@itsjeyd The wiki page is supposed to be the source-of-truth; it's what Axim refers to when we are updating or auditing a CC's access rights in GitHub. If it's out of date then that's something we should fix! Do you see any examples on that page that need fixing?

Comment (itsjeyd) [reply migrate/labels.yml:118]: @kdmccormick 

> The wiki page is supposed to be the source-of-truth; it's what Axim refers to when we are updating or auditing a CC's access rights in GitHub.

Ah OK 💡 Good to know. 

> If it's out of date then that's something we should fix! Do you see any examples on that page that need fixing?

I can't point out any specific examples in terms of things that are out of date (seeing as I don't know each and every CC's repo access levels 😅).

A more general question would be: Is there a defined process for making sure that the contents of the page are up-to-date?

In terms of concrete things to fix:

* The list of repos is not always exhaustive. Example (Jenna):
    > Modular Learning, Product Core, **many others**

* ... Or it's not clear where to find a detailed list of repos if needed. Example (Viktor, Matjaz):
    > Paragon Blended Support and Maintenance

    > All repos currently in the latest Named Release

* Sometimes it's not clear what types of access a CC has. Example (Michelle):
    > All repos pull requests!

Comment (mphilbrick211) [reply migrate/labels.yml:118]: Agreed with @itsjeyd - @kdmccormick sometimes that wiki page only lists areas of focus/expertise, and not the repo(s) the CC has rights in. It would be great if it DID list that!

Comment (kdmccormick) [reply migrate/labels.yml:118]: Yes, updating it is part of our CC onboarding & offboarding runbooks: https://openedx.atlassian.net/wiki/spaces/COMM/pages/3359342743/Participant+Administrator+Runbooks

Thanks for that feedback. Non-coding CCs are not granted write access to any repositories, so their "Projects Worked On" column is more informational than authoritative; we should make that clearer. As for coding CCs with rights like "All repos currently in the latest Named Release", that does mean something specific to Axim engineers, but I can see how it's opaque to others.

I wonder if we can avoid any ambiguity by linking you directly to permission listings on GitHub. @itsjeyd , can you see either of these pages, or do you get a 404?
* https://github.com/orgs/openedx/teams/frontend-all
* https://github.com/orgs/openedx/people/brian-smith-tcril

Comment (mphilbrick211) [reply migrate/labels.yml:118]: @kdmccormick FWIW, I can see the frontend-all one, but not Brian's

Comment (kdmccormick) [reply migrate/labels.yml:118]: Good to know @mphilbrick211 !

Comment (kdmccormick) [reply migrate/labels.yml:118]: @mphilbrick211 @itsjeyd I have done my best to clarify the page:
* I added some "Notes on Projects and Access"
* I broke up each contributor's access categories using bullet points
* Wherever a group of repositories were referred to, I linked to the corresponding GitHub team, which lists out the exact repos in the group 

I hope that helps!

FYI @sarina (on PTO currently)

Comment (sarina) [reply migrate/labels.yml:118]: This seems like a good addition that enough people have commented on already 🙂 

Is there anything specific you'd like me to chime in on, or just an inform?

Comment (kdmccormick) [reply migrate/labels.yml:118]: Just an inform!

Comment (itsjeyd) [reply migrate/labels.yml:118]: > I have done my best to clarify the page: ...

Thanks a lot @kdmccormick, that's a great improvement! 🚀 

> I wonder if we can avoid any ambiguity by linking you directly to permission listings on GitHub. @itsjeyd , can you see either of these pages, or do you get a 404?
> 
> * https://github.com/orgs/openedx/teams/frontend-all
> * https://github.com/orgs/openedx/people/brian-smith-tcril

Same as @mphilbrick211, frontend-all works fine but brian-smith-tcril 404s for me.

File: migrate/labels.yml
Comment (kdmccormick) [reply migrate/labels.yml]: Good call, I like that site better anyway, thanks!

File: migrate/labels.yml
Comment (kdmccormick) [reply migrate/labels.yml]: Perfect, I'll add that in.

File: migrate/labels.yml
Comment (kdmccormick) [reply migrate/labels.yml]: @nedbat For the time being, the bot will still apply a few of labels to PRs (just `blended` and `open-source-contribution` if I read [labels.py](https://github.com/openedx/openedx-webhooks/blob/master/openedx_webhooks/labels.py) correctly), even though it won't sync all of [labels.yml](https://github.com/openedx/openedx-webhooks-data/blob/main/labels.yaml) any more. I'm not sure if we actually need to define `blended` here in order for that to work, but I'll do it anyway just for completeness.

File: migrate/labels.yml
Comment (kdmccormick) [reply migrate/labels.yml]: After thinking about this, I ended up landing on `code health`.  I think that without a separate label, this stuff would end up lumped in with `maintenance`.

I almost went with `refactoring`, but Ned's comment got me thinking that we should avoid just re-implementing conventional commits here. The (imagined) purpose of these labels is to broadly categorize issues and PRs, and both `maintenance` and `code health` PRs are likely to have a mix of `feat`, `fix`, `refactor`, and `chore` commits depending on exactly what type of code change is required.

**Comment by kdmccormick** [comment]
Thank @wittjeff , I went and added `a11y` as a label. I don't see the other labels used in any of our repositories yet, so I imagine that they being used in 2U's Jira right now, yeah? Feel free to add those `WCAG*` labels to individual repos as you need, and if you find that you're using them in GitHub regularly, then let's add them at the organization level here.

**Comment by kdmccormick** [comment]
Comments are addressed; I think this is ready to go. Please let me know by EOD tomorrow (Thurs) if you have any concerns!

**Comment by kdmccormick** [comment]
Applying now...

**Comment by kdmccormick** [comment]
Almost done, but hit GitHub's rate limit. Will finish later.

**Pull request merged**
